protocol/Telemetry.proto

```diff
@@ -0,0 +1,200 @@
+ *
+ * Conditions:
+ *  - MQTT connection established, re-established
+ *  - Peripheral connected, disconnected
+ */
+message Registration {
+    /**
+     * Firmware version identifier.
+     */
+    uint32 version = 1;
+
+    /**
+     * Microcontroller unique identifier (MAC address).
+     */
+    bytes uuid = 2;
+
+    /**
+     * Microcontroller IP address.
+     */
+    bytes ip = 3;
+
+    /**
+     * List of connected peripherals.
+     */
+    repeated Peripheral_t peripherals = 4;
+    message Peripheral_t {
+        uint32 busId = 1;
+        uint32 busAddr = 2;
+        bytes generalCallResp = 3;
+    }
+}
+
+/**
+ * Provisioning frame (backend -> microcontroller).
+ *
+ * Conditions:
+ *  - Received Registration frame from microcontroller.
+ *  - Received new read definitions from corresponding peripheral controller.
+ */
+message Provisioning {
+    /**
+     * Default bus address for peripheral.
+     */
+    uint32 busAddr = 1;
+
+    /**
+     * Display name of peripheral processor.
+     */
+    string name = 2;
+
+    /*
+     * Peripheral read definitions.
+     */
+    repeated ReadDef_t readDefinitions = 3;
+    message ReadDef_t {
+        /**
+         * An arbitrary ID for external reference. This does not relate to the
+         * hardware or any application logic in the I2CRuntime. This is used
+         * externally to track read definitions.
+         */
+        uint32 definitionId = 1;
+
+        /**
+         * Some peripherals have 16-bit register IDs and some have 8-bit register
+         * IDs.
+         */
+        RegLength_t registerIdLength = 2;
+        enum RegLength_t {
+            RL16 = 0;
+            RL8 = 1;
+        }
+
+        /**
+         * Data is read on this output from a contiguous block of register IDs.
+         * e.g., 0x80 to 0xFF. This is the first register ID of the block.
+         */
+        uint32 registerId = 3;
+
+        /**
+         * Data is read on this output from a contiguous block of register IDs.
+         * e.g., 0x80 to 0xFF. This defines the number of register IDs in the block.
+         * This is essentially how many times the loop will read bytes at a register
+         * and then advance to the next register. For many peripherals this value is
+         * simple 1 (i.e., there is no need to advance).
+         */
+        uint32 registerBlockLength = 4;
+
+        /**
+         * The number of bytes that will be read at each register ID in the
+         * contiguous block. This means that the total number of bytes retrieved
+         * from one ReadDefinition instance is:
+         * 
+         *     numBytesPerRegister * registerBlockLength
+         */
+        uint32 numBytesPerRegister = 5;
+
+        /**
+         * How many milliseconds between reading all bytes from the block of
+         * registers?
+         */
+        uint32 readPeriod = 6;
+    }
+}
+
+/**
+ * Payload frame (microcontroller -> backend).
```

Comment:
Include a comment here on why we don't bother putting anymore identifying information here (e.g., which microcontroller) because that will come from elsewhere?